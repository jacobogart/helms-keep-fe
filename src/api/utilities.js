export const host = 'http://localhost:5000/api/';

export const mockTask = {
  id: '1234',
  isCompleted: false, 
  message: 'Clean the kitchen'
};

export const mockCompletedTask = {
  id: '5678',
  isCompleted: true,
  message: 'Make a mock completed task'
}

export const mockQuest = {
	id: '1234',
  title: 'My First Quest',
  color: "green",
	challenges: [ mockTask, mockCompletedTask ]
};
