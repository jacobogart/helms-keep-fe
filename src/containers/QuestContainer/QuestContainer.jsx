import React, { Component } from 'react';
import Quest from '../Quest/Quest';
import { connect } from 'react-redux';

export class QuestContainer extends Component {
  
  render() {
    const { quests, results } = this.props;
    let questsToRender = results 
      ? results.map(result => quests.find(quest => quest.id === result))
      : quests;

    const questCards = questsToRender.map(quest => (
      <Quest data={quest} key={quest.id} />
    ));
		return <section className="QuestContainer">{questCards}</section>;
	}
}

export const mapStateToProps = state => ({
  quests: state.quests
})

export default connect(mapStateToProps)(QuestContainer);
