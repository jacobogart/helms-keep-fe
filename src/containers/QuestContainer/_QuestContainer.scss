.QuestContainer {
  display: flex;
  flex-wrap: wrap;
  overflow: scroll;
  height: calc(100vh - 215px);
}
