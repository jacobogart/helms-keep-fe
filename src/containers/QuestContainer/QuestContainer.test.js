import React from 'react';
import { shallow } from 'enzyme';
import { QuestContainer, mapStateToProps } from './QuestContainer';
import { mockQuest } from '../../api/utilities';

describe('QuestContainer', () => {
	let wrapper;

	beforeEach(() => {
		wrapper = shallow(
		<QuestContainer 
			quests={[mockQuest, mockQuest]}
			results={null}
		/>);
	});

	it('should match snapshot', () => {
		expect(wrapper).toMatchSnapshot();
	});

	it('should match snapshot with results', () => {
		wrapper = shallow(
			<QuestContainer
				quests={[mockQuest, mockQuest]}
				results={['1234']}
			/>);
		expect(wrapper).toMatchSnapshot();
	});

	describe('MSTP', () => {
		it('should return an array of quests', () => {
			const state = { quests: [mockQuest, mockQuest], test: 'test' }
			const expected = { quests: [mockQuest, mockQuest] }
			const result = mapStateToProps(state);
			expect(result).toEqual(expected);
		});
	});
});
