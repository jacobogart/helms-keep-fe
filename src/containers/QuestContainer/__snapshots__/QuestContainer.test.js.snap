// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestContainer should match snapshot 1`] = `
<section
  className="QuestContainer"
>
  <ConnectFunction
    data={
      Object {
        "challenges": Array [
          Object {
            "id": "1234",
            "isCompleted": false,
            "message": "Clean the kitchen",
          },
          Object {
            "id": "5678",
            "isCompleted": true,
            "message": "Make a mock completed task",
          },
        ],
        "color": "green",
        "id": "1234",
        "title": "My First Quest",
      }
    }
    key="1234"
  />
  <ConnectFunction
    data={
      Object {
        "challenges": Array [
          Object {
            "id": "1234",
            "isCompleted": false,
            "message": "Clean the kitchen",
          },
          Object {
            "id": "5678",
            "isCompleted": true,
            "message": "Make a mock completed task",
          },
        ],
        "color": "green",
        "id": "1234",
        "title": "My First Quest",
      }
    }
    key="1234"
  />
</section>
`;

exports[`QuestContainer should match snapshot with results 1`] = `
<section
  className="QuestContainer"
>
  <ConnectFunction
    data={
      Object {
        "challenges": Array [
          Object {
            "id": "1234",
            "isCompleted": false,
            "message": "Clean the kitchen",
          },
          Object {
            "id": "5678",
            "isCompleted": true,
            "message": "Make a mock completed task",
          },
        ],
        "color": "green",
        "id": "1234",
        "title": "My First Quest",
      }
    }
    key="1234"
  />
</section>
`;
